pkg/controller/history/controller_history.go

```diff
@@ -244,20 +247,25 @@ func (rh *realHistory) CreateControllerRevision(parent metav1.Object, revision *
 	}
 	clone := any.(*apps.ControllerRevision)
 
+	// First attempt to create the revision will not use collisionCount
```

Comment:
> Are the differences intentional?

@janetkuo and I were working on DaemonSet history and StatefulSet history concurrently. The diverse implementation was a result of concurrency under time pressure and not specialization of the use case.

> Is this code meant to eventually be used generically by others controllers like DS too?

Yes. We always wanted to converge implementation, but, imo, that can be addressed in a follow up PR after giving it more consideration.